alica_engine/include/engine/BasicPlan.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+namespace alica
+{
+class BasicPlan {
```

Comment:
So since `init` and `onTermination` are mutable, a BasicPlan shouldn't be owned by the model, i think i misunderstood that earlier. `RunningPlanExtension`? `BasicPlanImpl`? `PlanStateNotifier`? Not sure...

Either way, a `Plan` cannot know about them given their mutablility. Instead they probably should be fully owned by the corresponding RunningPlan.